src/smdba/postgresqlgate.py

```diff
@@ -903,14 +903,22 @@ def do_system_check(self, *args, **params):
         conf_path = self.config['pcnf_pg_data'] + "/postgresql.conf"
         conf = self._get_conf(conf_path)
         changed = False
+        max_connections = 400
 
         #
         # Setup postgresql.conf
         #
 
         # Built-in tuner
+        if params.get('max_connections') != None:
+            max_connections = int (params.get('max_connections'))
+
+        if max_connections < 400:
+            print >> sys.stdout, "INFO: max_connections should be at least 400."
+            max_connections = 400
+
         if 'autotuning' in args:
-            for item, value in PgTune().estimate().config.items():
+            for item, value in PgTune(max_connections).estimate().config.items():
```

Comment:
No. If the admin passes a parameter that is smaller than 400, it should print a warning and set the parameter to the minimum of 400.